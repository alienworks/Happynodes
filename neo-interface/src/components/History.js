import React, { Component } from 'react'
import { connect } from 'react-refetch'
import config from './config'
import { AreaChart, Area, XAxis, YAxis, CartesianGrid, Tooltip } from 'recharts';
import ReactResizeDetector from 'react-resize-detector';

class HistoryTest extends Component {
    render() {
        const { historytest } = this.props;


        if (historytest.pending) {
            return (
                <div className="col col-lg-12 col-md-12 col-sm-12 col-xs-12">
                    <div className="jumbotron">
                        <h2>Loading...</h2>
                    </div></div>
            )
        } else if (historytest.rejected) {

            return (
                <div className="col col-lg-12 col-md-12 col-sm-12 col-xs-12">
                    <div>
                        error
                </div></div>
            )
        } else if (historytest.fulfilled) {
            let rows = historytest.value['data'];

            const SpecialAreaChart = ({ width, height }) => (

                <AreaChart width={width} height={400} data={rows}
                    margin={{ top: 10, right: 30, left: 0, bottom: 0 }}>
                    <CartesianGrid strokeDasharray="3 3" />
                    <XAxis dataKey="date" />
                    <YAxis />
                    <Tooltip />
                    <Area type='monotone' dataKey='total' stroke='#8884d8' fill='#8884d8' />
                    <Area type='monotone' dataKey='totalonline' stroke='#82ca9d' fill='#82ca9d' />
                </AreaChart>

            );


            return (
                <div className="col col-lg-12 col-md-12 col-sm-12 col-xs-12">
                    <div className="jumbotron nodes">

                        <h2>Historic View: Online and Total Nodes Daily</h2>
                        <hr style={{
                            'borderColor': '#78cadd', 'borderWidth': '3px'
                        }} />

                        <h3>We released V1.1.0 (Node Discovery) at the start of August which is why the number of nodes we know about increased significantly after that.</h3>


                        <ReactResizeDetector handleWidth handleHeight>
                            <SpecialAreaChart />
                        </ReactResizeDetector>


                    </div></div>

            )
        }
    }
}



export default connect((props) => ({
    historytest: {
        url: config.api_url.concat(`/historic/network/size/daily`)
    }
}))(HistoryTest)