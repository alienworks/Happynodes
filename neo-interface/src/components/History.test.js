import React from 'react';
import ReactDOM from 'react-dom';
import History from './History';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<History />, div);
    ReactDOM.unmountComponentAtNode(div);
});

var data ={"data":[{"date":"2018-06-14","totalonline":37,"total":45},{"date":"2018-06-15","totalonline":37,"total":45},{"date":"2018-06-16","totalonline":37,"total":45},{"date":"2018-06-17","totalonline":40,"total":45},{"date":"2018-06-18","totalonline":38,"total":44},{"date":"2018-06-19","totalonline":37,"total":45},{"date":"2018-06-20","totalonline":37,"total":45},{"date":"2018-06-21","totalonline":37,"total":45},{"date":"2018-06-22","totalonline":39,"total":45},{"date":"2018-06-23","totalonline":37,"total":43},{"date":"2018-06-24","totalonline":38,"total":45},{"date":"2018-06-25","totalonline":36,"total":45},{"date":"2018-06-26","totalonline":38,"total":45},{"date":"2018-06-27","totalonline":38,"total":45},{"date":"2018-06-28","totalonline":36,"total":43},{"date":"2018-06-29","totalonline":36,"total":45},{"date":"2018-06-30","totalonline":36,"total":45},{"date":"2018-07-01","totalonline":36,"total":45},{"date":"2018-07-02","totalonline":36,"total":45},{"date":"2018-07-03","totalonline":35,"total":39},{"date":"2018-07-04","totalonline":40,"total":45},{"date":"2018-07-05","totalonline":38,"total":43},{"date":"2018-07-06","totalonline":40,"total":45},{"date":"2018-07-07","totalonline":34,"total":39},{"date":"2018-07-08","totalonline":37,"total":45},{"date":"2018-07-09","totalonline":38,"total":45},{"date":"2018-07-10","totalonline":43,"total":50},{"date":"2018-07-11","totalonline":43,"total":50},{"date":"2018-07-12","totalonline":42,"total":50},{"date":"2018-07-13","totalonline":42,"total":50},{"date":"2018-07-14","totalonline":43,"total":50},{"date":"2018-07-15","totalonline":43,"total":50},{"date":"2018-07-16","totalonline":43,"total":50},{"date":"2018-07-17","totalonline":43,"total":50},{"date":"2018-07-18","totalonline":40,"total":50},{"date":"2018-07-19","totalonline":42,"total":50},{"date":"2018-07-20","totalonline":44,"total":50},{"date":"2018-07-21","totalonline":42,"total":50},{"date":"2018-07-22","totalonline":40,"total":50},{"date":"2018-07-23","totalonline":40,"total":50},{"date":"2018-07-24","totalonline":43,"total":50},{"date":"2018-07-25","totalonline":42,"total":50},{"date":"2018-07-26","totalonline":41,"total":50},{"date":"2018-07-27","totalonline":41,"total":50},{"date":"2018-07-28","totalonline":43,"total":50},{"date":"2018-07-29","totalonline":41,"total":50},{"date":"2018-07-30","totalonline":99,"total":118},{"date":"2018-07-31","totalonline":98,"total":128},{"date":"2018-08-01","totalonline":100,"total":130},{"date":"2018-08-02","totalonline":91,"total":127},{"date":"2018-08-03","totalonline":89,"total":123},{"date":"2018-08-04","totalonline":96,"total":132},{"date":"2018-08-05","totalonline":93,"total":134},{"date":"2018-08-06","totalonline":96,"total":134},{"date":"2018-08-07","totalonline":97,"total":136},{"date":"2018-08-08","totalonline":97,"total":137},{"date":"2018-08-09","totalonline":97,"total":139},{"date":"2018-08-10","totalonline":97,"total":139},{"date":"2018-08-11","totalonline":92,"total":138},{"date":"2018-08-12","totalonline":93,"total":140},{"date":"2018-08-13","totalonline":101,"total":149},{"date":"2018-08-14","totalonline":93,"total":150},{"date":"2018-08-15","totalonline":100,"total":159},{"date":"2018-08-16","totalonline":99,"total":159}]}
describe("test against valid snapshot", ()=> {
    it("redner as expected", ()=> {
        const tree = renderer.create(<History historytest={{fulfilled:true, value:data}} />).toJSON();
        expect(tree).toMatchSnapshot();

    });
});
  