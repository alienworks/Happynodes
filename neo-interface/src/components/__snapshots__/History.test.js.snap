// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test against valid snapshot redner as expected 1`] = `
<div
  className="col col-lg-12 col-md-12 col-sm-12 col-xs-12"
>
  <div
    className="jumbotron nodes"
  >
    <h2>
      Historic View: Online and Total Nodes Daily
    </h2>
    <hr
      style={
        Object {
          "borderColor": "#78cadd",
          "borderWidth": "3px",
        }
      }
    />
    <h3>
      We released V1.1.0 (Node Discovery) at the start of August which is why the number of nodes we know about increased significantly after that.
    </h3>
    <div
      style={
        Object {
          "display": "none",
          "height": 0,
          "position": "absolute",
          "visibility": "hidden",
          "width": 0,
        }
      }
    />
  </div>
</div>
`;
